libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -227,16 +227,31 @@ void AP_Baro::calibrate(bool save)
    update the barometer calibration
    this updates the baro ground calibration to the current values. It
    can be used before arming to keep the baro well calibrated
+
+   The altitude argument is an optional argument holding the current absolute
+   altitude of the vehicle. If it is set to 0, then we assume there is no
```

Comment:
Good catch, will update the comment.

I started by passing in the desired altitude from the outside caller. But it was resulting in a *lot* of code duplication, as every caller had to request the gps fix status, ensure it was good enough, extract the altitude, convert the unit and pass it along. In the end this was proving to be a lot messier then passing the GPS directly in. And since the GPS is our only source of absolute altitude information at the moment, I could't justify the complexity of passing the altitude.